### ISSUE TITLE
flux-tree: rollup the exit codes to the top of the tree

### ISSUE THREAD

**Issue by dongahn** [flux-framework/flux-core#2695]
Peeling off a request from @SteVwonder from Issue #521: https://github.com/flux-framework/flux-sched/issues/512#issuecomment-532907220

**Comment by dongahn** [comment]
Yes, I was going to use the parent's guest namespace to roll up the exit code of each and every jobscript being launched at the leaf node for `flux-tree`. 

I thought about providing this as part of flux-core. But that probably requires a fully featured "reap" service where we rollup the job provenance data of each nested instance to the top?

```bash
FLUX_KVS_NAMESPACE=$(flux getattr parent-kvs-namespace) \
FLUX_URI=$(flux getattr parent-uri) \
flux kvs put -j "${script_index}.exit_code=${return_code}"
```

**Comment by garlick** [comment]
How does this case differ from the way we want batch jobs to work in a system instance?

A follow on question is what is missing from our current model for batch jobs?  We need issues open for the missing bits...

**Comment by dongahn** [comment]
> A follow on question is what is missing from our current model for batch jobs? We need issues open for the missing bits...

I believe the key missing piece here is the "reap" operation which reaps the job information from each and every one of the nested instance and copy (or aggregate) it to the parent level.

**Comment by dongahn** [comment]
Sorry. I thought I opened this as a flux-sched issue as @grondo said. I misread his comment.
